pkg/builtin/multi/lpad.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multi
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/matrixorigin/matrixone/pkg/builtin"
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend/overload"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/lpad"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+var ArgAndRets_Lpad = []argsAndRet{
+	{[]types.T{types.T_varchar, types.T_int64, types.T_varchar}, types.T_varchar},
+}
+
+func init() {
+	extend.FunctionRegistry["lpad"] = builtin.Lpad
+	for _, item := range ArgAndRets_Lpad {
+		overload.AppendFunctionRets(builtin.Lpad, item.args, item.ret)
+	}
+	extend.MultiReturnTypes[builtin.Lpad] = func(es []extend.Extend) types.T {
+		return getMultiReturnType(builtin.Lpad, es)
+	}
+
+	extend.MultiStrings[builtin.Lpad] = func(es []extend.Extend) string {
+		return fmt.Sprintf("lpad(%s, %s, %s)", es[0], es[1], es[2])
+	}
+	overload.OpTypes[builtin.Lpad] = overload.Multi
+	overload.MultiOps[builtin.Lpad] = []*overload.MultiOp{
+		{
+			Min:        3,
+			Max:        3,
+			Typ:        types.T_varchar,
+			ReturnType: types.T_varchar,
+			Fn: func(vecs []*vector.Vector, proc *process.Process, cs []bool) (*vector.Vector, error) {
+				vs := vecs[0].Col.(*types.Bytes) //Get the first arg
+
+				if !cs[1] && vecs[1].Typ.Oid != types.T_int64 {
```

Comment:
I think the logical operator here is **OR**

Reply:
I tried mysql, and I think mysql accepts both constant value and attribute name as lpad's parameter, in that case, we don't need this check
```go
!cs[1]
```

```go
!cs[2]
```

Reply:
I think we need different function implementations to handle this hairy constant scenario, namely:
 parameter1 | parameter2 | parameter3
 ----------- |  ----------- | ----------
 constant   |    constant.  |   constant
constant.   |    constant.   |  column 
constant.   |    column      | column
...... |


the builtin power function has some neat examples you may refer to